esp-hal/src/spi/master.rs

```diff
@@ -1022,12 +1121,46 @@ where
     )]
     /// or is below 70kHz,
     /// [`ConfigError::UnsupportedFrequency`] error will be returned.
+    ///
+    /// ## Example
+    ///
+    /// ```rust, no_run
+    #[doc = crate::before_snippet!()]
+    /// # use esp_hal::spi::Mode;
+    /// # use esp_hal::spi::master::{Config, Spi};
+    /// let mut spi = Spi::new(peripherals.SPI2, Config::default())?;
+    ///
+    /// spi.apply_config(&Config::default()
+    /// .with_frequency(Rate::from_khz(100)));
```

Comment:
The formatting is a bit off here, we should format examples like rustfmt would format them.

Reply:
In an application this would be: 
```rust
spi.apply_config(&Config::default().with_frequency(Rate::from_khz(100)));
```

But due to the docstring (`///`) is too long and `rustfmt` format it as: 
```rust
/// spi.apply_config(&Config::default().
/// with_frequency(Rate::from_khz(100)));
```
Rendered:
![image](https://github.com/user-attachments/assets/a6ebdd41-f11b-46ba-acd1-7b18d05a095f)

Which is weird